applications/plugins/BeamPlastic/forceField/BeamPlasticFEMForceField.inl

```diff
@@ -0,0 +1,2530 @@
+                                        "indicates if a precomputed elastic stiffness matrix is used, instead of being computed by reduced integration"))
+    , d_useConsistentTangentOperator(initData(&d_useConsistentTangentOperator, false, "useConsistentTangentOperator", 
+                                             "indicates wether to use a consistent tangent operator in the computation of the plastic stiffness matrix"))
+    , d_isPerfectlyPlastic(initData(&d_isPerfectlyPlastic, false, "isPerfectlyPlastic", "indicates wether the behaviour model is perfectly plastic"))
+    , d_modelName(initData(&d_modelName, std::string("RambergOsgood"), "modelName", "the name of the 1D contitutive law model to be used in plastic deformation"))
+    , d_zSection(initData(&d_zSection, (Real)zSection, "zSection", "length of the section in the z direction for rectangular beams"))
+    , d_ySection(initData(&d_ySection, (Real)ySection, "ySection", "length of the section in the y direction for rectangular beams"))
+    , d_useSymmetricAssembly(initData(&d_useSymmetricAssembly,false,"useSymmetricAssembly","use symmetric assembly of the matrix K"))
+    , d_isTimoshenko(initData(&d_isTimoshenko, isTimoshenko, "isTimoshenko", "implements a Timoshenko beam model"))
+    , m_edgeHandler(NULL)
+{
+    m_edgeHandler = new BeamFFEdgeHandler(this, &m_beamsData);
+
+    d_poissonRatio.setRequired(true);
+    d_youngModulus.setReadOnly(true);
+}
+
+template<class DataTypes>
+BeamPlasticFEMForceField<DataTypes>::~BeamPlasticFEMForceField()
+{
+    if (m_edgeHandler) delete m_edgeHandler;
+}
+
+
+/*****************************************************************************/
+/*                           INITIALISATION                                  */
+/*****************************************************************************/
+
+
+template <class DataTypes>
+void BeamPlasticFEMForceField<DataTypes>::bwdInit()
+{
+    core::behavior::BaseMechanicalState* state = this->getContext()->getMechanicalState();
+    assert(state);
+    m_lastUpdatedStep=-1.0;
+}
+
+template <class DataTypes>
+void BeamPlasticFEMForceField<DataTypes>::init()
+{
+    this->core::behavior::ForceField<DataTypes>::init();
+    sofa::core::objectmodel::BaseContext* context = this->getContext();
+
+    m_topology = context->getMeshTopology();
+
+    m_stiffnessContainer = context->core::objectmodel::BaseContext::get<container::StiffnessContainer>();
+    m_poissonContainer = context->core::objectmodel::BaseContext::get<container::PoissonContainer>();
+
+    // Retrieving the 1D plastic constitutive law model
+    std::string constitutiveModel = d_modelName.getValue();
+    if (constitutiveModel == "RambergOsgood")
+    {
+        Real youngModulus = d_youngModulus.getValue();
+        Real yieldStress = d_yieldStress.getValue();
+        fem::RambergOsgood<DataTypes> *RambergOsgoodModel = new (fem::RambergOsgood<DataTypes>)(youngModulus, yieldStress);
```

Comment:
If you keep handling more than one constitutive models, please use a sd::unique_ptr instead of a raw pointer. I guess you have a memory leak here...